jukeos/src/components/Player.js

```diff
@@ -69,10 +138,70 @@ const Player = ({ children }) => {
     const syncTrack = () => {
       player.getCurrentState().then((state) => {
         if (state && state.track_window.current_track) {
+          console.log(track?.uri !== state.track_window.current_track.uri);
+          if (track?.uri !== state.track_window.current_track.uri) {
+            fetchRecentlyPlayed();
+          }
           setTrack(state.track_window.current_track);
-          console.log("Setting Track: " + JSON.stringify(track));
+          console.log("Setting Track: " + track?.uri);
```

Comment:
Could probably comment this out. This is left by me so thats on me:
```suggestion
          //console.log("Setting Track: " + track?.uri);
```